spec/editors_draft.html

```diff
@@ -3298,6 +3298,244 @@ <h2>Evaluation Semantics</h2>
 
   </section>
 
+  <section id="import-export-composite-values">
+    <h1>Importing and Exporting of Composite Values</h1>
+
+    <section class="informative">
+      <h2>Motivation</h2>
+      <p>
+        In the normal handling of an RDF syntax, <a data-cite="RDF11-CONCEPTS#dfn-blank-node-identifier">blank node identifiers</a> are scoped to the current document, and any identifiers used in that document are merely a syntactic convenience for referencing blank nodes freshly allocated during parsing.
+        As an example, consider the following snippet of Turtle syntax [[TURTLE]].
+      </p>
+        
+      <pre id="load-renaming-blank-example"
+         data-transform="updateExample"
+         data-content-type="text/turtle"
+         class="nohighlight example">
+      <!--
+      _:b1 ex:p4 "[ <http://example.org/alice>, <http://example.org/bob>, _:eve ]"^^cdt:List .
+      -->
+      </pre>
+      
+      <p>
+        Parsing this data as Turtle will result in a single triple whose subject is a blank node (now dissociated from the `b1` identifier), and whose object is a <a>cdt:List literal</a> that represents a three-element <a>term list</a> in which the third element is a blank node.
+        However, using a Turtle parser not aware of the <a href="#list-datatype">cdt:List datatype</a> will result in an object literal containing the substring "_:eve"; that is, the literal will keep the exact lexical form as given in the Turtle snippet above (<a href="#load-renaming-blank-example"></a>).
+        Allowing multiple such composite values, from multiple sources, to be inserted into the same RDF Dataset (e.g., via two SPARQL `LOAD` updates) presents a challenge:
+        when a system interprets the literals obtained from its Turtle parser, the <a>term lists</a> that it creates by applying the <a href="#list-datatype-lex-to-value-mapping">cdt:List lexical-to-value mapping</a> will contain the same blank nodes (because, during the mapping process for each cdt:List literal, the <i><a>bnl2bn</a></i> function is applied to the same blank node identifier, `_:eve`).
```

Comment:
While this text focuses on the case of a system that is aware of our CDT datatypes but uses a Turtle parser that is not (i.e., the system interprets the CDT literals after receiving them from the Turtle parser), there is also another issue that we don't mention here; namely for systems that do not support the cdt:List datatype at all. In the case of the example outlined in the text here, such a system would throw all the cdt:List literals from different sources together, without renaming `_:eve` in any of these literals. Hence, piping data through such a system and then into a cdt:List-aware system may cause differences in query results, compared to loading the data directly into the cdt:List-aware system. In other words, a system that is not aware of the cdt:List datatype may accidentally change the meaning of cdt:List literals collected from multiple sources (by making these literals all contain the same blank nodes). Do you agree that this is a problem?

Reply:
So, you are suggesting we add a Note to call out this issue?

Even if users of CDTs are aware of it, I am afraid that there is nothing that they will be able to do to avoid the problem if they combine using systems that are and that are not CDT aware. That's somewhat unsatisfying, but I also don't see another way to address this issue (except for disallowing blank nodes in CDT literals).

I should also mention that this issue was one of the concerns raised in discussions of the approach at ESWC.

Reply:
Yeah, understood. Possible paths forward are to have bnode support be optional, with clear documentation that supporting them would introduce incompatibility with non-CDT-supporting systems; or to just document the issue with strong warnings about using bnodes if you need interop between systems…